CountingSort/app_resources/prefix_sum_shader.comp.hlsl

```diff
@@ -0,0 +1,53 @@
+#include "nbl/builtin/hlsl/bda/__ptr.hlsl"
+#include "nbl/builtin/hlsl/sort/counting.hlsl"
+#include "nbl/builtin/hlsl/sort/common.hlsl"
+
+[[vk::push_constant]] nbl::hlsl::sort::CountingPushData pushData;
+
+struct PtrAccessor
+{
+    static PtrAccessor createAccessor(uint64_t addr)
+    {
+        PtrAccessor ptr;
+        ptr.addr = addr;
+        return ptr;
+    }
+
+    uint32_t get(uint64_t index)
+    {
+        return bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().load();
+    }
+
+    void set(uint64_t index, uint32_t value)
+    {
+        bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().store(value);
+    }
+
+    bda::__spv_ptr_t<uint32_t> get_ptr(uint64_t index)
+    {
+        return bda::__ptr < uint32_t > (addr + sizeof(uint32_t) * index).template
+        deref().get_ptr();
+    }
```

Comment:
`atomicAdd` instead to keep with accessors being storage space and underlying type agnostic

Reply:
Done